aptos-move/framework/aptos-token-objects/sources/property_map.move

```diff
@@ -215,34 +201,36 @@ module aptos_token_objects::property_map {
     // Accessors
 
     public fun contains_key<T: key>(object: &Object<T>, key: &String): bool acquires PropertyMap {
-        assert!(
-            exists<PropertyMap>(object::object_address(object)),
-            error::not_found(EPROPERTY_MAP_DOES_NOT_EXIST),
-        );
+        check_property_map_exists(object::object_address(object));
         let property_map = borrow_global<PropertyMap>(object::object_address(object));
         simple_map::contains_key(&property_map.inner, key)
     }
 
     public fun length<T: key>(object: &Object<T>): u64 acquires PropertyMap {
-        assert!(
-            exists<PropertyMap>(object::object_address(object)),
-            error::not_found(EPROPERTY_MAP_DOES_NOT_EXIST),
-        );
+        check_property_map_exists(object::object_address(object));
         let property_map = borrow_global<PropertyMap>(object::object_address(object));
         simple_map::length(&property_map.inner)
     }
 
+    /// Read the property and get it's external type in it's bcs encoded format
+    ///
+    /// The preferred method is to use `read_<type>` where the type is already known.
     public fun read<T: key>(object: &Object<T>, key: &String): (String, vector<u8>) acquires PropertyMap {
-        assert!(
-            exists<PropertyMap>(object::object_address(object)),
-            error::not_found(EPROPERTY_MAP_DOES_NOT_EXIST),
-        );
+        check_property_map_exists(object::object_address(object));
         let property_map = borrow_global<PropertyMap>(object::object_address(object));
         let property_value = simple_map::borrow(&property_map.inner, key);
         let new_type = to_external_type(property_value.type);
         (new_type, property_value.value)
     }
 
+    inline fun check_property_map_exists(object: address) {
```

Comment:
in the interest of being anal, we are already in `property_map` so the additional namespacing doesn't add value, we could just consider
`assert_exists`?

check feels more like something I'd say should return a bool